### ISSUE TITLE
Fullscreen makes input unresponsive

### ISSUE THREAD

**Issue by eternalistic** [jdolan/quake2#17]
I just upgraded to macOS High Sierra and when I open Quake it drops me to the console and is unresponsive to any keyboard input.

I did a clean install by compiling Quake using the `make` and `make install` commands. Everything succeeded and I was able to open Quake just fine. Then I changed my video settings in the game to "Custom" and enabled fullscreen and saved my changes. When the game reloaded with the new settings the keyboard was unresponsive and I couldn't bring up any menu or enter any commands.

I deleted my existing install of Quake and reinstalled so I could go back through the process and troubleshoot what was going on. This time I changed my video settings to "Custom" and left fullscreen disabled and saved the changes. The game reloaded and everything was working correctly. I was able to launch the single player and play just fine. Then I tried the fullscreen option again and once that was enabled and the game reloaded the keyboard was again unresponsive.

I'd be happy to do some more troubleshooting if anyone has something I could try to do to fix this. Not sure what else to try at the moment.

**Comment by jdolan** [comment]
I won't be able to upgrade to High Sierra anytime too soon to test this. I can confirm that the game still works on Sierra.

What happens if you try:

`./quake2 +set r_customwidth 0 +set r_customheight 0 +set gl_mode -1`

That should attempt to use your desktop resolution. If that fails, try setting other reasonable values for `r_customwidth` and `r_customheight` and re-launching.

**Comment by eternalistic** [comment]
I was able to run the suggested command and the game fires up in windowed mode and is fullscreen size. So it mostly shows on the screen and which is what I was able to successfully run earlier. As soon as I switch it to fullscreen in the video settings it reloads the game and drops me back to the console and all input is unresponsive.

The game is definitely playable in the windowed mode so this will just have to do for now until we can figure out how to get it working in fullscreen mode again.

I'll keep messing with it and see if there is something else I'm missing here. Thanks!

**Comment by jdolan** [comment]
Just curious.. did you rebuild libSDL the other day when you also rebuilt libPNG?

**Comment by jdolan** [comment]
Also, are you open for contract work? I'm looking for a Drupal developer right now at my day job: http://cantina.co

**Comment by eternalistic** [comment]
No, I just rebuilt libPNG. If you think rebuilding libSDL would be worth a shot I can certainly try that and see how it goes. The only frustrating part is after making changes and enabling fullscreen there is no keyboard input so I have to reboot my computer and try something else again.

Sent you an invite on LinkedIn so we can connect.

**Comment by eternalistic** [comment]
I was able to get an error report after the game crashed when trying to switch to fullscreen mode.

```
Using /System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib for OpenGL...setting mode 13: 1280 1024
GL_VENDOR: ATI Technologies Inc.
GL_RENDERER: AMD Radeon R9 M370X OpenGL Engine
GL_VERSION: 2.1 ATI-1.58.30
...GL_EXT_compiled_vertex_array not found
...GL_EXT_point_parameters not found
...using GL_ARB_multitexture
...using GL_EXT_texture_rectangle
...using GL_SGIS_generate_mipmap
...ignoring GL_ARB_texture_compression
...enabling GL_EXT_texture_filter_anisotropic
Maximum Anisotropy: 16.0
Maximum Texture Size: 16384x16384
------------------------------------
libcurl/7.50.3 OpenSSL/1.0.2j zlib/1.2.8 libidn/1.17 initialized.
====== Quake II (Quetoo.org) Initialized ======

--------- [Loading Renderer] ---------
dlopen(./libGL.so, 1): image not found
R_Init: Could not load "libGL.so", trying "/System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib"
Using /System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib for OpenGL...setting mode 13: 1280 1024
GL_VENDOR: ATI Technologies Inc.
GL_RENDERER: AMD Radeon R9 M370X OpenGL Engine
GL_VERSION: 2.1 ATI-1.58.30
...GL_EXT_compiled_vertex_array not found
...GL_EXT_point_parameters not found
...using GL_ARB_multitexture
...using GL_EXT_texture_rectangle
...using GL_SGIS_generate_mipmap
...ignoring GL_ARB_texture_compression
...enabling GL_EXT_texture_filter_anisotropic
Maximum Anisotropy: 16.0
Maximum Texture Size: 16384x16384
------------------------------------
Abort trap: 6
```

Seems like this is the problem:
```
--------- [Loading Renderer] ---------
dlopen(./libGL.so, 1): image not found
R_Init: Could not load "libGL.so", trying "/System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib"
Using /System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib for OpenGL...setting mode 13: 1280 1024
```

I tried searching for a copy of the `libGL.so` file but am unable to locate it. As mentioned in #10, I tried to symlink it to another file `/usr/lib/x86_64-linux-gnu/libGLU.so.1` but that file doesn't exist either. And the Quake 2 app doesn't include the file in the `Content/MacOS/lib` directory.

**Comment by jdolan** [comment]
That output is actually entirely expected on OS X. On Windows, "standard operating procedure" is to check the current working directory for shared objects (`.dll` files). Quake2 tries the same approach on all platforms, but you'll never find `libGL.so` on Mac (nor in the Quake2 folder on Linux). The "fallback" is to go find the System's version of OpenGL, which it does successfully per your log.

However, your log does end with something interesting:

`Abort trap: 6`

This is the signal handler, which means that the crash should be entirely debuggable. Can you run:

```
lldb ./Contents/MacOS/quake2
run
```

When the game crashes, hopefully you can Cmd-Tab back to the terminal and type:

```
bt
```

To yield a backtrace.

**Comment by eternalistic** [comment]
I have Quake installed in `/Users/[my-username]/.quake2` so I opened Terminal and ran `lldb ./quake2` and hit enter a couple of times to accept the changes. 

Then I ran Quake by typing `./quake2` and that opened the game. I then enabled fullscreen and it went back to the console but this time it was accepting keyboard input. I started a singeplayer game and everything was running just fine. So I closed the game and re-opened and it was still in fullscreen mode and all working still.

Seems to be working correctly now. Thanks!
